packages/sol-cov/src/trace.ts

```diff
@@ -0,0 +1,76 @@
+import { OpCode, StructLog, TransactionTrace } from '@0xproject/types';
+import { BigNumber, logUtils } from '@0xproject/utils';
+import { addHexPrefix, stripHexPrefix } from 'ethereumjs-util';
+import * as fs from 'fs';
+import * as _ from 'lodash';
+
+export interface TraceByContractAddress {
+    [contractAddress: string]: StructLog[];
+}
+function padZeros(address: string): string {
+    return addHexPrefix(_.padStart(stripHexPrefix(address), 40, '0'));
+}
+
+export function getTracesByContractAddress(structLogs: StructLog[], startAddress: string): TraceByContractAddress {
+    const traceByContractAddress: TraceByContractAddress = {};
+    let currentTraceSegment = [];
+    const callStack = [startAddress];
+    // tslint:disable-next-line: prefer-for-of
+    for (let i = 0; i < structLogs.length; ++i) {
+        const structLog = structLogs[i];
+        if (structLog.depth !== callStack.length - 1) {
+            throw new Error("Malformed trace. trace depth doesn't match call stack depth");
+        }
+        // After that check we have a guarantee that call stack is never empty
+        // If it would: callStack.length - 1 === structLog.depth === -1
+        // That means that we can always safely pop from it
+        currentTraceSegment.push(structLog);
+
+        if (
+            structLog.op === OpCode.CallCode ||
+            structLog.op === OpCode.StaticCall ||
+            structLog.op === OpCode.Call ||
+            structLog.op === OpCode.DelegateCall
+        ) {
+            const currentAddress = _.last(callStack) as string;
+            const jumpAddressOffset = structLog.op === OpCode.DelegateCall ? 4 : 2;
```

Comment:
magic numbers.

Reply:
Not really. I can't describe it more than `jumpAddressOffset`. Those numbers come from yellowpaper. I could describe what are other variables on the stack during that call but I doubt that's useful.

Reply:
Let's add a comment about 4 & 2 coming from yellowpaper, and add `Source: ` link to the PDF.